pkg/scheduler/apis/config/types_pluginargs.go

```diff
@@ -166,6 +166,17 @@ type NodeResourcesMostAllocatedArgs struct {
 	Resources []ResourceSpec
 }
 
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// NodeResourcesBalancedAllocationArgs holds arguments used to configure NodeResourcesBalancedAllocation plugin.
+type NodeResourcesBalancedAllocationArgs struct {
+	metav1.TypeMeta
+
+	// Resources to be considered when scoring.
+	// The default resource set includes "cpu" and "memory".
+	Resources []WeightlessResource
```

Comment:
thanks, `WeightlessResource` is still a struct even it only has one field, this will make it consistent with other node resource plugins regarding to configuration.
Just let me know if you want this to be a `[]string`.

Reply:
we use to do this in that way: https://github.com/kubernetes/kubernetes/pull/101946#discussion_r631822090,  pls check the comments here.

Reply:
This type looks awkward. We don't save much by avoiding to validate that the weight is 0.

Reply:
yeah, an alternative is to default 0 to 1, and validate that weight is only 1. I'm fine with that too.

Reply:
Done